src/Sentry/Internal/Hub.cs

```diff
@@ -230,6 +258,18 @@ public void CaptureTransaction(Transaction transaction)
             }
         }
 
+        public void CaptureSession(SessionUpdate sessionUpdate)
+        {
+            try
+            {
+                _ownedClient.CaptureSession(sessionUpdate);
```

Comment:
Do we have `CaptureSession` in the spec?
I was hoping this would not be exposed. The client can take the Session from the scope (that can be internal) and add to an envelope if one was provided. I believe we do that for attachments too, no?

Reply:
Attachments are not sent separately from events, but sessions are. So we need this as part of the interface.

Reply:
Another language limitation/design issue?

We don't need users to capture sessions, so it ideally stays an internal business. The API is: `StartSessions/EndSessions`.

Can we have an internal extension method to capture that which will wrap in an envelope?

Or as I originally suggested, `Scope.Session`, we just set the Session instance to the scope before passing down to the client (I believe I did that on iOS)

Reply:
It's not a language limitation, just the fact that `ISentryClient` is public, which it is (I assume) to let users provide their own instance in tests. Users don't directly interact with `ISentryClient`, but they need it to be public. And because it's public, these methods need to be public to.

If we change `ISentryClient` to deal with envelopes directly, I fail to see how its purpose going to be different than what's already managed by `ITransport` (which works with envelopes directly).

> Or as I originally suggested, Scope.Session, we just set the Session instance to the scope before passing down to the client (I believe I did that on iOS)

Yeah, we do it.